src/mosquito/backend.cr

```diff
@@ -9,7 +9,11 @@ module Mosquito
     end
 
     def self.build_key(*parts)
-      KeyBuilder.build KEY_PREFIX, *parts
+      if global_prefix = Mosquito.configuration.global_prefix
+        KeyBuilder.build global_prefix, KEY_PREFIX, *parts
+      else
+        KeyBuilder.build KEY_PREFIX, *parts
+      end
```

Comment:
Wonderfully done @dammer. Do you think there's a need for both KEY_PREFIX and global_prefix? I can't decide 🤔 

If not, it might suffice to simply change what KEY_PREFIX itself means, or how it is sourced:

```crystal
   # KEY_PREFIX becomes a getter or class getter:
   getter key_prefix : String { Mosquito.configuration.global_prefix }

   # Or just ditch key_prefix altogether and read from configuration:
   def self.build_key(*parts)
      KeyBuilder.build Mosquito.configuration.global_prefix, *parts
   end

   # and in Configuration:
   property key_prefix : String = "mosquito"
```

Reply:
I simplified my solution a bit by adding `Nil` handling in the `KeyBuilder`, which allows removing the condition in `build_key` and simplifies the solution slightly. I also removed the direct use of `mosquito` without key_build.

Reply:
UPD: I noticed an inconsistency when running the PR on the master branch: the periodic job key is not prefixed. Take a look at `"mosquito:periodic_jobs:StatusJob"`, all other keys are okay. Could it be that the Coordinator runs before the configuration?

```log
127.0.0.1:6379[5]> MONITOR
OK
1715696179.556765 [5 127.0.0.1:34060] "select" "5"
1715696179.556848 [5 127.0.0.1:34062] "select" "5"
1715696179.556977 [5 127.0.0.1:34060] "script" "load" "  if redis.call(\"get\",KEYS[1]) == ARGV[1] then\n      return redis.call(\"del\",KEYS[1])\n  else\n      return 0\n  end"
1715696179.557012 [5 127.0.0.1:34062] "script" "load" "  if redis.call(\"get\",KEYS[1]) == ARGV[1] then\n      return redis.call(\"del\",KEYS[1])\n  else\n      return 0\n  end"
1715696179.557088 [5 127.0.0.1:34060] "set" "ryhNEB9tD2K8pvZDdNZKnT:mosquito:coordinator:football" "0e494ddf5ac9522d" "ex" "10" "nx"
1715696179.557137 [5 127.0.0.1:34062] "zrange" "ryhNEB9tD2K8pvZDdNZKnT:mosquito:queues" "0" "-1"
1715696179.557374 [5 127.0.0.1:34076] "select" "5"
1715696179.557636 [5 127.0.0.1:34076] "hget" "mosquito:periodic_jobs:StatusJob" "last_executed_at"
1715696179.557880 [5 127.0.0.1:34062] "hset" "ryhNEB9tD2K8pvZDdNZKnT:mosquito:job_run:1715696179557:8" "enqueue_time" "1715696179557" "type" "status_job" "retry_count" "0"
1715696179.557981 [5 127.0.0.1:34076] "lpush" "ryhNEB9tD2K8pvZDdNZKnT:mosquito:waiting:status_job" "1715696179557:8"
1715696179.558000 [5 127.0.0.1:34076] "zadd" "ryhNEB9tD2K8pvZDdNZKnT:mosquito:queues" "1715696179" "status_job"
1715696179.558063 [5 127.0.0.1:34062] "hset" "mosquito:periodic_jobs:StatusJob" "last_executed_at" "1715696179"

```

Reply:
Yep it is early than configuration https://github.com/mosquito-cr/mosquito/blob/703a90a71d22c5e0495633e0af1c0b5bbd606c26/src/mosquito/periodic_job.cr#L24